### PR TITLE
refactor: use `ParYield` in `RedBlackTree`

### LINKED ISSUE #4867: Use ParYield in Standard Library (e.g. RedBlackTree)

### PR DESCRIPTION
**Pull request by jaschdoc**
Should wait for #4913 
Closes #4867

### CODE REVIEW

**Comment by magnus-madsen** [comment]
@jaschdoc Can I merge this even though something is broken or?

**Comment by jaschdoc** [comment]
Are you referring to https://github.com/flix/flix/issues/4917 ?

File: main/src/library/RedBlackTree.flix [hunk 1]
```diff
@@ -378,23 +378,18 @@ namespace RedBlackTree {
...
-                    else {
-                        spawn (Channel.send(parExists((n - 2) / 2, f, a), sL)); // We divide the rest of the threads as follows:
-                        spawn (Channel.send(parExists((n - 2) / 2, f, b), sR)); // We spawn two new threads leaving us with n - 2
-                        let a1 = Channel.recv(rL);                              // that we distribute over the two spanned threads.
-                        let b1 = Channel.recv(rR);
-                        a1 or b1
-                    }
+                    else
+                        par (left <- parExists((n - 2) / 2, f, a); // We divide the rest of the threads as follows:
+                            right <- parExists((n - 2) / 2, f, b)) // We spawn two new threads leaving us with n - 2
```
Comment (magnus-madsen) [inline main/src/library/RedBlackTree.flix:391]: @mlutze and I know what you did....

Comment (magnus-madsen) [reply main/src/library/RedBlackTree.flix:391]: You could not help yourself, you really wanted those arrows to align :D

File: main/src/library/RedBlackTree.flix [hunk 2]
```diff
@@ -646,22 +636,18 @@ namespace RedBlackTree {
...
-                    let (sR, rR) = Channel.buffered(1);
-                    spawn (Channel.send(parMapWithKey((n - 2) / 2, f, a), sL)); // We divide the rest of the threads as follows:
-                    spawn (Channel.send(parMapWithKey((n - 2) / 2, f, b), sR)); // We spawn two new threads leaving us with n - 2
-                    let v1 = f(k, v);                                           // that we distribute over the two spanned threads.
-                    let a1 = Channel.recv(rL);
-                    let b1 = Channel.recv(rR);
-                    Node(c, a1, k, v1, b1)
+                    par (left <- parMapWithKey((n - 2) / 2, f, a); // We divide the rest of the threads as follows:
+                        right <- parMapWithKey((n - 2) / 2, f, b); // We spawn two new threads leaving us with n - 2
+                        this  <- f(k, v))                          // that we distribute over the two spanned threads.
```
[the remark this answers is not in the document]
Comment (jaschdoc) [reply main/src/library/RedBlackTree.flix:649]: should it just be `val`?

File: main/src/library/RedBlackTree.flix (hunk 1 above)
Comment (jaschdoc) [reply main/src/library/RedBlackTree.flix:391]: haha yeah but it's also the default indentation! :wink:

File: main/src/library/RedBlackTree.flix (hunk 2 above)
[the remark this answers is not in the document]
Comment (magnus-madsen) [reply main/src/library/RedBlackTree.flix:649]: Sounds good

File: main/src/library/RedBlackTree.flix (hunk 1 above)
Comment (magnus-madsen) [reply main/src/library/RedBlackTree.flix:391]: how about `l` and `r`?

**Comment by magnus-madsen** [comment]
- [ ] Conflicts (with Matt's apologies; bad timing).
- [ ] left -> l, right -> r
- [ ] this -> v or nv or something. Just not this or val or some Scala keyword :D

**Comment by magnus-madsen** [comment]
> > Any thoughts on indentation of `yield`?
> 
> Here's the way I would do it:
> 
> ```
> par (
>     x <- e1;
>     y <- e2;
> ) yield (x, y)
> ```
> 
> Which matches Scala style

Yes, but if there are few elements (like two) its OK to keep it on the same line IMO

**Comment by jaschdoc** [comment]
Alright done

**Comment by jaschdoc** [comment]
Yeah very

**Pull request merged**